api/V1/User.js

```diff
@@ -24,38 +25,55 @@ module.exports = function(app, baseUrl) {
    *
    * @apiUse V1ResponseError
    */
-  app.post(apiUrl, function(req, res) {
-    if (!req.body.username || req.body.username == 'undefined' ||
-      !req.body.password || req.body.password == 'undefined') {
-      return responseUtil.send(res, {
-        statusCode: 400,
-        success: false,
-        messages: ['Missing username or password.']
+  app.post(
+    apiUrl,
+    middleware.logging,
+    middleware.parseParams({
+      body: {
+        username: { type: 'STRING' },
+        password: { type: 'STRING' },
+      },
```

Comment:
Rather than writing our code to do parameter validation what about using this? https://www.npmjs.com/package/express-validator

There's a number of other similar packages, but this looks like one of the best ones to me (based only a quick look though)